src/Assignments/SimpleProblemPage.tsx

```diff
@@ -70,8 +70,20 @@ export const SimpleProblemPage: React.FC<SimpleProblemPageProps> = () => {
 
         const currentTopic = res.data.data.topic;
 
-        if (currentTopic.studentTopicOverride?.length > 0) {
-            _.assign(currentTopic, currentTopic.studentTopicOverride[0]);
+        // apply topic overrides: start/end/dead dates
+        if (!_.isNil(currentTopic.studentTopicOverride) && !_.isEmpty(currentTopic.studentTopicOverride)) {
+            const override = currentTopic.studentTopicOverride[0];
+            delete override.id;
+            _.assign(currentTopic, override);
+        }
+        // apply assessment overrides: duration, maxAttempts, maxVersions, versionDelay
+        // only maxVersions is relevant -- the others are accounted for in version (studentTopicAssessmentInfo)
+        if (!_.isNil(currentTopic.topicAssessmentInfo) && 
+            !_.isNil(currentTopic.topicAssessmentInfo.studentTopicAssessmentOverride) && 
+            !_.isEmpty(currentTopic.topicAssessmentInfo.studentTopicAssessmentOverride)
```

Comment:
isEmpty checks nil, I don't know if strict will accept it as a nil check as well though, in theory you might not need the isNil check

Reply:
This is NIT btw

Reply:
it didn't pass without the additional check, believe me - I'd rather this were shorter